### ISSUE TITLE
Introduce `Set.AddAll(iter.Seq)`

### ISSUE THREAD

**Issue by jbduncan** [jbduncan/go-containers#21]
Blocked on the currently-experimental [range-over-func](https://github.com/golang/go/issues/61405) Go feature being merged into a stable version of Go.

Introduce `Set.AddAll(iter.Seq)`, where `iter.Seq` is a type alias for a range-able function. This will let any container type or iterator type be used as the argument.

Inspired by proposal https://github.com/golang/go/discussions/47331.
